tv_apps/tv_shared/js/card_manager.js

```diff
@@ -629,32 +652,16 @@
       return found;
     },
 
-    // TODO: need to be protected by state and _reloadCardList
-    findFolderFromCardList: function cm_findFolderFromCardList(query) {
-      var found;
-      this._cardList.some(function(card) {
-        if (card instanceof Folder && card.folderId === query.folderId) {
-          found = card;
-          return true;
-        }
-      });
-      return found;
-    },
-
     isPinned: function cm_isPinned(options) {
       var that = this;
       return this._getPipedPromise('isPinned', function(resolve, reject) {
-        // we only answered isPinned when card manager is in READY state
-        if (that.isReady()) {
-          resolve(!!that.findCardFromCardList(options));
-        } else {
-          // card manager is not ready, so we append query at the end of
-          // promise of _reloadCardList (which would guarantee card manager will
-          // be READY eventually)
-          that._reloadCardList().then(function() {
-            resolve(!!that.findCardFromCardList(options));
-          });
-        }
+        that._asyncSemaphore.wait(function() {
+          resolve();
```

Comment:
you can combine the `that._reloadCardList()` below here as `resolve(that._reloadCardList())`.